src/components/menu/menu-item/menu-item.component.tsx

```diff
@@ -201,71 +192,33 @@ export const MenuItem = ({
   }, [registerItem, unregisterItem]);
 
   useEffect(() => {
-    const inputIcon = ref.current?.querySelector(
-      "[data-element='input-icon-toggle']",
-    );
-    if (!openSubmenuId && focusFromSubmenu === undefined && focusFromMenu) {
-      /* istanbul ignore else */
-      if (focusRef.current) {
-        (focusRef.current as HTMLElement)?.focus();
-      }
-    } else if (
-      focusFromSubmenu &&
-      !(shiftTabPressed && inputIcon?.getAttribute("tabindex") === "0")
-    ) {
-      /* istanbul ignore else */
-      if (focusRef.current) {
-        (focusRef.current as HTMLElement)?.focus();
-      }
-    }
-  }, [
-    openSubmenuId,
-    focusFromMenu,
-    focusFromSubmenu,
-    shiftTabPressed,
-    focusRef,
-  ]);
+    if ((focusFromMenu && !focusFromSubmenu) || focusFromSubmenu)
```

Comment:
question: are these changes within the scope of this commit (i.e. upgrading React to v18) as well?

Reply:
Relates to: https://github.com/Sage/carbon/pull/7148#discussion_r1925577335